enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,517 @@
+History`.
+
+## Drawbacks
+
+The idea is to find the best form of an argument why this enhancement should _not_ be implemented.
+
+## Alternatives
+
+The best approach to achieve the goals described above is not clear,
+so the alternatives section is a collection of candidates for
+discussion.
+
+### Expressing a need to scale using an API
+
+These alternatives discuss what the API would look like, without
+considering which CRD it would be on.
+
+#### A CPU resource request size parameter
+
+We could define an API to give a general size parameter, using values
+like t-shirt sizes (large, medium, small) or flavors.
+
+Such an API is vague, and it would be up to teams to interpret the
+size value appropriately. We would also be limited in the number of
+sizes we could reasonably describe to users before it would become
+confusing (see Amazon EC2 flavors).
+
+#### A CPU resource "request limit" parameter
+
+We could define an API to give an absolute limit of the CPU resource
+requests for the cluster services.
+
+Operators would need to scale the resource requests for their operands
+based on an understanding of how their defaults were derived. For
+example, today most values are a proportion of the resources needed
+for etcd on a 3 node control plane where each node has 2 CPUs, for a
+total of 6 CPUs. If the API indicates a limit of 2 CPUs, the operator
+would need to configure its operands to request 1/3 of the default
+resources.
+
+#### A CPU resource scaling parameter
+
+We could define an API to give a scaling parameter that operators
+should use for their operands.
+
+The value would be applied globally based on understanding how the
+defaults are set for 6 CPUs and how many CPUs are available in the
+real system. We could document the 6 CPU starting point to allow users
+to make this calculation themselves.
+
+Operators could read the setting and use it for their operands, but
+not themselves.
+
+### Locations for API setting
+
+These alternatives could apply to any of the API forms discussed in
+the previous section.
+
+#### Use a cluster-wide API
+
+We could add a field to the Infrastructure configuration
+resource. Many of our operators, including OLM-based operators,
+already consume the Infrastructure API.
+
+#### Use an operator-specific API
+
+If we choose to implement a new operator to manage scaling for the
+control plane operators (see below), it could define a new API in a
+CRD it owns.
+
+Updating existing operators to read the new field and scale their
+operands and documenting an entirely new API might be more work than
+using the Infrastructure API that would only be useful if the operator
+responding to the API was an add-on.
+
+### Dynamic vs. Static Setting
+
+It is not likely that users would need to change the setting after
+building a cluster, so we could have the installer set a status field
+in whatever CRD we choose for the API. If we do need to support
```

Comment:
Would the chain be user -> installer -> static setting?  Because teaching the installer to automatically know what the setting should be strikes me as harder than teaching some in-cluster component to automatically know what the setting should be (because an in-cluster component could actually look, while the installer will have to guess ahead of time).  If the installer is just forwarding a user choice, that seems achievable, but we'd need to train users in selecting an appropriate value.

Reply:
Yes, I was thinking a new field in install-config.

Reply:
I added some text to clarify that the setting would come from the installer's input and not need to be derived.

Reply:
Mark this thread resolved so it doesn't distract future reviewers, now that it's been addressed in the text?

Reply:
Yep, I wanted to get an ack from you before resolving it.